nodedb_test.go

```diff
@@ -248,3 +244,128 @@ func makeHashes(b *testing.B, seed int64) [][]byte {
 	b.StartTimer()
 	return hashes
 }
+
+func TestNodeDB_SaveFastNode_ReturnsErrorForNilKey(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockNode := NewMockNode(ctrl)
+	mockNode.EXPECT().getKey().Return(nil).Times(1)
+
+	ndb, _, _, _ := createNodeDbWithMocks(ctrl)
+	err := ndb.SaveFastNode(mockNode)
+	require.Error(t, err)
+}
+
+func TestNodeDB_SaveFastNode_FailsWhenUnableToWriteToBuffer(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockNode := NewMockNode(ctrl)
+	var buf bytes.Buffer
+	mockNode.EXPECT().getKey().Return([]byte("k1")).Times(1)
+	mockNode.EXPECT().EncodedSize().Return(1).Times(1)
+	mockNode.EXPECT().WriteBytes(gomock.AssignableToTypeOf(&buf)).Return(fmt.Errorf("error")).Times(1)
+
+	ndb, _, _, _ := createNodeDbWithMocks(ctrl)
+	err := ndb.SaveFastNode(mockNode)
+	require.Error(t, err)
+}
+
+// TODO: We should have explicit argument types for below functions
```

Comment:
What does this comment mean? Do we need to test it with different implementations of `Node` interface? Can we do that?

Reply:
No I amended the comment to be clear. Basically in places where we have goMock.Any(), we ideally should have the explicitly argument expected so that our unit tests are clearly stating what is expected. That way if something changes the tests fail to reflect that rather then possibly being a false pass